### PR TITLE
Misc cleanup

### PR DESCRIPTION
**Pull request by zrax**
... plus demystification of dialog resources

### CODE REVIEW

File: chips.asm [hunk 1]
```diff
@@ -430,80 +430,131 @@ ALIGN SectorSize, db 0
...
-    db 0x09, 0x00, 0x07, 0x00, 0x84, 0x00, 0x13, 0x00, 0xff, 0xff, 0x00, 0x00, 0x02, 0x50, 0x82, "Enter a level number and password, or just a password.", 0, 0
-    db 0x45, 0x00, 0x1e, 0x00, 0x20, 0x00, 0x0c, 0x00, 0x64, 0x00, 0x80, 0x00, 0x81, 0x50, 0x81, 0, 0
-    db 0x45, 0x00, 0x2e, 0x00, 0x20, 0x00, 0x0c, 0x00, 0x65, 0x00, 0x88, 0x00, 0x81, 0x50, 0x81, 0, 0
-    db 0x10, 0x00, 0x20, 0x00, 0x33, 0x00, 0x08, 0x00, 0xff, 0xff, 0x02, 0x00, 0x02, 0x50, 0x82, "Level number:", 0, 0
-    db 0x10, 0x00, 0x2f, 0x00, 0x33, 0x00, 0x08, 0x00, 0xff, 0xff, 0x02, 0x00, 0x02, 0x50, 0x82, "Password:", 0, 0
-    db 0x1b, 0x00, 0x4a, 0x00, 0x28, 0x00, 0x0e, 0x00, ID_OK, 0x0, 0x01, 0x00, 0x01, 0x50, 0x80, "OK", 0, 0
-    db 0x53, 0x00, 0x4a, 0x00, 0x28, 0x00, 0x0e, 0x00, ID_CANCEL,0,0x00, 0x00, 0x01, 0x50, 0x80, "Cancel", 0, 0
+    DIALOG 6, 18, 151, 94, 7, {"Go To Level",0}, 8, {"MS Sans Serif",0}, DS_MODALFRAME|DS_SETFONT
+
+    STATICTEXT  {"Enter a level number and password, or just a password.",0}, -1, 9, 7, 132, 19
```
Comment (magical) [inline chips.asm]: I couldn't find STATICTEXT in the resource compiler docs. Should this be LTEXT? https://docs.microsoft.com/en-us/windows/win32/menurc/ltext-control

Also the trailing NULs in the item text should be in the macro, not here.

**Review by magical — CHANGES_REQUESTED** [review]
Hmm, I think the dialog sections could be clearer.

For the DIALOG itself, each parameter used to be on a separate line with a nice comment explaining what it was; now they're just a jumble of numbers on one line -- you have to look at the macro definition to figure out what they are.   That seems like a step backwards. And there are so many dialog macros now that i'm tempted to move them to a separate file, which would increase the distance between definition and use even further.

For the dialog items, the structure looked regular enough that i was considering a single DLGITEM macro along with judicious use of constants. Are you trying to mimic the resource compiler syntax? I guess that could work. I'd still like to make it more obvious to the reader what each field is, through comments or alignment or both.

(Long term i'd like to have an actual rc file that defines these but obviously we're not there yet.)

File: chips.asm (hunk 1 above)
Comment (zrax) [reply chips.asm]: > I couldn't find STATICTEXT in the resource compiler docs. Should this be LTEXT? https://docs.microsoft.com/en-us/windows/win32/menurc/ltext-control

Yes, I was being lazy and combining the three TEXT types into a single one...  I like your idea of moving the macros to a separate file though, so I can do that and be less lazy.


> Also the trailing NULs in the item text should be in the macro, not here.

I agree.  Unfortunately, I couldn't find a way to have the macro only add the trailing NUL if the string itself was non-null (several of the STATICTEXT objects have a null text entry), and I didn't really like the idea of adding a separate macro just for TEXT controls with null text.  Suggestions welcome :)

EDIT: Of course now that I think about it more, it should be possible to just put an empty `""` string...  I'm too used to thinking like C :/

**Review by magical — APPROVED** [review]
Looks a lot better, thanks.

I see you figured out the empty string trick ;)

**Pull request merged**